balancer/frontend-v3
# feat: Allow slippage on forced proportional adds

Allow the user to set a custom slippage in forced proportional add liquidity flows. Setting this slippage adjusts the user's balances in TokenBalancesProvider so that they can only input their balance - set slippage.

### How to test
- In a CoW pool add liquidity flow, try changing the slippage setting to something greater than 0%.
  - Notice your balances changing?
  - Try adding liquidity, how much dust are you left with?
- In a non-forced proportional pool, try adding liquidity to see if anything has been affected.

<img width="316" alt="Screenshot 2024-10-02 at 09 47 36" src="https://github.com/user-attachments/assets/e45e9dcc-6a35-4658-aec0-10f53112af5b">
<img width="363" alt="Screenshot 2024-10-02 at 09 47 43" src="https://github.com/user-attachments/assets/bb50dd8e-02e5-4d6d-a140-c10768b591a3">
<img width="492" alt="Screenshot 2024-10-02 at 09 47 49" src="https://github.com/user-attachments/assets/4aa1774c-c65a-462a-ab32-d6762eaacfab">
